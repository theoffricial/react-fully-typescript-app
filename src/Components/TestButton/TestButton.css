.test-button {
                                                                                border: 1px
                                                                                                                                                                solid
                                                                                                                                                                green;
                                                                                padding: 10px;
}
