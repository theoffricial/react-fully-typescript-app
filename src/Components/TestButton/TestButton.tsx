import React from 'react';
import "./TestButton.css";

interface TestButtonProps {
    label: string;
}

const TestButton = (props: TestButtonProps): JSX.Element => {
    return <button data-testid="test-button" className="test-button">{props.label}</button>
}

export default TestButton;