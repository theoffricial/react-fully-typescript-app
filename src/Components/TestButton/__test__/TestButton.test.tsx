import React from 'react';
import TestButton from "../TestButton";
import renderer from 'react-test-renderer';
it("<TestButton /> renders correctly", () => {
    const component = renderer.create(<TestButton label="test"></TestButton>);
    expect(component.root.props).toEqual({ label: "test" });
    
    const button = component.toJSON() as renderer.ReactTestRendererJSON;
    expect(button).toMatchSnapshot();
});