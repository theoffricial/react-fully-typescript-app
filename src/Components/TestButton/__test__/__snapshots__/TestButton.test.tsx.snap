// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestButton /> renders correctly 1`] = `
<button
  className="test-button"
  data-testid="test-button"
>
  test
</button>
`;
